### PR TITLE
Added ordinals support for time block

### LINKED ISSUE #1323: Add ordinals support to time block
Hello,

I was wondering if it were possible to get ordinals support in the time block? Currently, I'm using the following custom block:
```
date '+%A %eXX %B, %H:%M' | sed -e 's/11XX/11th/' -e 's/12XX/12th/' -e 's/13XX/13th/' -e 's/1XX/1st/' -e 's/2XX/2nd/' -e 's/3XX/3rd/' -e 's/XX/th/'
```
Which gives me `Wednesday 27th October, 12:58` but I was hoping to get this built into the existing block.

### PR DESCRIPTION
**Pull request by virajbhartiya**
Fixed #1323 

I've added Ordinal support for time using `chrono`

### CODE REVIEW

**Comment by MaxVerevkin** [comment]
FYI, `pre-commit` can be run locally (`pre-commit install`)

**Comment by virajbhartiya** [comment]
Thanks @MaxVerevkin, I test pre-commit locally then commit

**Comment by MaxVerevkin** [comment]
@virajbhartiya I believe you pushed to the wrong branch.

**Comment by virajbhartiya** [comment]
Yes, I as actually working on the padding issue but forgot the switch branches.
I'll fix it

**Comment by virajbhartiya** [comment]
Hey @MaxVerevkin, can you help me understand why the checks are failing, I haven't made any changes in the cargo.toml file but still it's showing errors in it

File: Cargo.toml [hunk 1]
```diff
@@ -43,6 +43,9 @@ smart-default = "0.6"
 swayipc-async = "2.0"
 tokio-stream = "0.1"
 toml = "0.7"
+chrono = "0.4"
```
Comment (bim9262) [inline Cargo.toml:46]: @virajbhartiya , it's unhappy that you have `chrono` and `chrono-tz` here becuase they are further down:

```toml
[dependencies.chrono]
version = "0.4"
default-features = false
features = ["clock", "unstable-locales"]

[dependencies.chrono-tz]
version = "0.8"
features = ["serde"]
```

File: Cargo.toml [hunk 2]
```diff
@@ -43,6 +43,7 @@ smart-default = "0.6"
 swayipc-async = "2.0"
 tokio-stream = "0.1"
 toml = "0.7"
+tera = "1.13"
```
Comment (MaxVerevkin) [inline Cargo.toml:46]: Why do you need `tera`?

File: src/blocks/time.rs [hunk 3]
```diff
@@ -86,7 +88,13 @@ pub async fn run(config: Config, mut api: CommonApi) -> Result<()> {
 
         widget.set_values(map!(
             "icon" => Value::icon(api.get_icon("time")?),
-            "timestamp" => Value::datetime(Utc::now(), timezone.copied())
+            "timestamp" => {
```
Comment (bim9262) [inline src/blocks/time.rs:91]: Looks like this will be a regression of #1723

File: src/blocks/time.rs [hunk 4]
```diff
@@ -86,7 +88,13 @@ pub async fn run(config: Config, mut api: CommonApi) -> Result<()> {
...
+        let timezone: Option<Tz> = None; // or parse the timezone from a string
+        let datetime: DateTime<Utc> = Utc::now();
+        let format: Option<&str> = Some("%a %e{S} %B, %H:%M");
+
+        Value::datetime(datetime, timezone, Some(format))
```
Comment (bim9262) [inline src/blocks/time.rs:96]: The third Parameter would be Option<Option<&str>>, which probably isn't what you want.

File: src/blocks/time.rs [hunk 5]
```diff
@@ -104,11 +112,3 @@ pub async fn run(config: Config, mut api: CommonApi) -> Result<()> {
         }
     }
 }
-
```
Comment (bim9262) [inline src/blocks/time.rs:107]: @MaxVerevkin we need this to be able to get the timezone if the user change `/etc/localtime`, right?

File: src/formatting/formatter.rs [hunk 6]
```diff
@@ -81,6 +82,12 @@ pub fn new_formatter(name: &str, args: &[Arg]) -> Result<Box<dyn Formatter>> {
                                 .error("Interval must be a positive number")?,
                         );
                     }
+                    "padding" => {
```
Comment (bim9262) [inline src/formatting/formatter.rs:85]: Looks like this code is part of the wrong PR.

**Comment by ammgws** [comment]
Not really sure what is happening here, but just to make sure - the original issue was to add support for formatting the number of the day of the month with the appropriate suffix such as `st`, `nd`, and `rd` (e.g. 1st, 2nd, 3rd).

Also the format string for the time block already had a breaking change in the upgrade to v0.30 so there shouldn't be anymore breaking changes especially not for a niche feature such as this one.

File: src/blocks/time.rs [hunk 7]
```diff
@@ -24,15 +24,17 @@
 //! block = "time"
 //! interval = 60
 //! [block.format]
-//! full = " $icon $timestamp.datetime(f:'%a %Y-%m-%d %R %Z', l:fr_BE) "
-//! short = " $icon $timestamp.datetime(f:%R) "
+//! full = " $icon $timestamp.datetime(f:'%a %e{S} %B, %H:%M', l:Europe/Lisbon) "
+//! short = " $icon $timestamp.datetime(f:%R, l:Europe/Lisbon) "
```
Comment (MaxVerevkin) [inline src/blocks/time.rs:28]: `l:` is for locale, not timezone

File: src/blocks/time.rs (hunk 5 above)
Comment (MaxVerevkin) [reply src/blocks/time.rs:107]: Yes. @virajbhartiya added `use libc::tzset`, but I can't find it in libc docs?

File: src/blocks/time.rs (hunk 7 above)
Comment (ammgws) [reply src/blocks/time.rs:28]: Why are these changes even necessary?

**Pull request closed without merging**